@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { DecodedTile } from "@here/harp-datasource-protocol";
+import { PerformanceTimer } from "@here/harp-utils";
+
+import { PerformanceStatistics } from "../Statistics";
+import { Tile } from "../Tile";
+import { TileGeometryCreator } from "./TileGeometryCreator";
+
+/**
+ * Loads the geometry for its [[Tile]]. Derived classes allow for different loading strategies.
+ */
+export interface TileGeometryLoader {
+    /**
+     * The [[Tile]] this `TileGeometryLoader` is managing.
+     */
+    tile: Tile;
+
+    /**
+     * `True` if all geometry of the `Tile` has been loaded and the loading process is finished.
+     */
+    isFinished: boolean;
+
+    /**
+     * `True` if the basic geometry has been loaded, and the `Tile` is ready  for display.
+     */
+    basicGeometryLoaded: boolean;
```

Comment:
what defines "basic" geometry?

Reply:
"`True` if the basic geometry has been loaded, and the `Tile` is ready  for display."
"basic" may differ based on the loading manager. Tiles with basic geometry may be ready for display, even while some parts (phases) are not yet loaded.